client/README.md

```diff
@@ -0,0 +1,54 @@
+
+## Client Package
+Client package implements network communication with the access nodes APIs. 
+It also defines a `Client` interface exposing all the common API interactions.  
+
+### Design
+Each implementation (currently `grpc` and `http`) include the following parts:
+- **Base Client** is the client that implements the client interface exposing 
+all generic functionality. 
+- **Client** is the network specific client that exposes any additional 
+options possible by the specific API implementation.
+- **Handler** takes care of actual network communication implementing 
+the communication protocol. 
+
+
+### Usage
+If you want to use the base client you should save the instance to 
+the client interface, which would allow you to easily switch between network 
+implementations like so:
+
+**General Usage**
+```go
+// common client interface
+var flowClient client.Client
+
+// initialize an http emulator client
+flowClient, err := http.NewDefaultEmulatorClient(false)
+
+// initialize a gPRC emulator client
+flowClient, err = grpc.NewDefaultEmulatorClient()
```

Comment:
> TBH I'm not a huge fan of having separate functions for each network; personally I think the host constants are enough.

Yeah, you are probably right. These factories are too thin to make sense. 

> Also, we use should the [mixed caps convention](https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps) instead of all-caps for those constants.

Nice, I actually prefer this, not sure why I thought the caps are idiomatic in Go.